### PR TITLE
Add example for ``bad-super-call``

### PR DESCRIPTION
**Pull request by DudeNr33**
Co-authored by: Vladyslav Krylasov <vladyslav.krylasov@gmail.com>

- [x] Write a good description on what the PR does.
- [x] If you used multiple emails or multiple names when contributing, add your mails
   and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|    | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Ref #5953

### CODE REVIEW

File: doc/data/messages/b/bad-super-call/details.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+In Python 2.7, ``super()`` has to be called with the own class and ``self`` as arguments (``super(Bar, self)``), which can
```
Comment (jacobtylerwalls) [inline doc/data/messages/b/bad-super-call/details.rst:1]:
```suggestion
In Python 2.7, ``super()`` has to be called with its own class and ``self`` as arguments (``super(Bar, self)``), which can
```

File: doc/data/messages/b/bad-super-call/details.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+lead to a mix up of parent and child class in the code.
+
+In Python 3 the recommended way is to user ``super()`` without arguments (see also ``super-with-arguments``).
```
Comment (jacobtylerwalls) [inline doc/data/messages/b/bad-super-call/details.rst:4]:
```suggestion
In Python 3 the recommended way is to call ``super()`` without arguments (see also ``super-with-arguments``).
```

Comment (DudeNr33) [reply doc/data/messages/b/bad-super-call/details.rst]: Thanks for the catch!

File: doc/data/messages/b/bad-super-call/bad.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class Foo:
+    pass
+
+
+class Bar(Foo):
+    def __init__(self):
+        super(Foo, self).__init__()  # [bad-super-call]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/b/bad-super-call/bad.py:7]:
```suggestion
class Animal:
    pass


class Cat(Animal):
    def __init__(self):
        super(Animal, self).__init__()  # [bad-super-call]
```

File: doc/data/messages/b/bad-super-call/details.rst [hunk 4]
```diff
@@ -0,0 +1,4 @@
+In Python 2.7, ``super()`` has to be called with its own class and ``self`` as arguments (``super(Bar, self)``), which can
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/b/bad-super-call/details.rst:1]:
```suggestion
In Python 2.7, ``super()`` has to be called with its own class and ``self`` as arguments (``super(Cat, self)``), which can
```

File: doc/data/messages/b/bad-super-call/good.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class Foo:
+    pass
+
+
+class Bar(Foo):
+    def __init__(self):
+        super().__init__()
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/b/bad-super-call/good.py:7]:
```suggestion
class Animal:
    pass


class Cat(Animal):
    def __init__(self):
        super().__init__()
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
I'm trying to avoid foo/bar in examples because it's not friendly for beginner, (the persons that are the most likely to read the documentation). https://github.com/PyCQA/pylint/pull/6115#issuecomment-1086561103

**Comment by DudeNr33** [comment]
That makes sense that this whole ``foobar`` thing might be confusing for beginners. Thanks for the review!

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**